lib/kafka/connection_builder.rb

```diff
@@ -1,25 +1,44 @@
+require 'kafka/sasl_gssapi_authenticator'
+
 module Kafka
   class ConnectionBuilder
-    def initialize(client_id:, logger:, instrumenter:, connect_timeout:, socket_timeout:, ssl_context:)
+    def initialize(client_id:, logger:, instrumenter:, connect_timeout:, socket_timeout:, ssl_context:, sasl_gssapi_principal:, sasl_gssapi_keytab:)
       @client_id = client_id
       @logger = logger
       @instrumenter = instrumenter
       @connect_timeout = connect_timeout
       @socket_timeout = socket_timeout
       @ssl_context = ssl_context
+      @sasl_gssapi_principal = sasl_gssapi_principal
+      @sasl_gssapi_keytab = sasl_gssapi_keytab
     end
 
     def build_connection(host, port)
-      Connection.new(
+      @connection = Connection.new(
         host: host,
         port: port,
         client_id: @client_id,
         connect_timeout: @connect_timeout,
         socket_timeout: @socket_timeout,
         logger: @logger,
         instrumenter: @instrumenter,
-        ssl_context: @ssl_context,
+        ssl_context: @ssl_context
       )
```

Comment:
The connection builder needs to be able to build _multiple_ connections, so we can't store a single connection in instance state.